### ISSUE TITLE
import inflow.dat slow

### ISSUE THREAD

**Issue by FLO-2DKaren** [FLO-2DSoftware/qgis-flo-2d-plugin#1610]
When you import inflow.dat using the import button on the BC Editor, it can take over 20 min to update the data.

This process is additive so it is not just clearing tables and then importing.  It's adding inflow.dat to cells that could already have inflow tables.

![image](https://github.com/user-attachments/assets/d38b34ab-6b51-43e0-8cf3-cf36cf44644b)

Test with this project.
https://flo-2d.sharefile.com/d-s449b5d9896e4403dba3e0ee7631736f2

**Comment by rpachaly** [comment]
Quick notes of the changes that I'm doing. Let me know if you agree. 

1. If a current BC in a grid has a TS, it will be updated not appended

**Comment by FLO-2DKaren** [comment]
I don't think that is a problem.  Do you think we would need that old data for any reason?

**Comment by FLO-2DKaren** [comment]
If I had a set of inflow nodes like a channel fill or something and I imported ones that were connected to a different line, would my channel inflow be deleted?  

I liked that the data was additive and not just a replacement.  That was very helpful that my inflow nodes weren't all wiped out.  

I think only clear data for the ones that are the same grid element number but leave old data if it isn't being replaced.

**Comment by rpachaly** [comment]
> If I had a set of inflow nodes like a channel fill or something and I imported ones that were connected to a different line, would my channel inflow be deleted?

I don't think I full understood what you mean, but I don't think they will be deleted. 

> I liked that the data was additive and not just a replacement. That was very helpful that my inflow nodes weren't all wiped out.

The data is still additive, unless importing data on a grid element that has BC. In this case, it would be updated.

**Comment by FLO-2DKaren** [comment]
OK I think that will be fine then.

**Comment by rpachaly** [comment]
I did some changes on the code to speed up. But the bottleneck was the database execute at each loop. I changed it to a execute_many and now it takes less than a minute in the project you sent. 

I also did some testing on other models and all seems to be fine. However, I don't feel totally comfortable merging this one before your testing. 

It is done on https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/pull/1615

**Comment by FLO-2DKaren** [comment]
I'll test it today.
